helpers/posts.js

```diff
@@ -77,61 +78,125 @@ exports.getPosts = async function (req, res, next) {
-            $inc: {
-                likes: 1
-            }
-        }
-    })
+    try{
+        post = await postModel.findByIdAndUpdate(
+            post_id,
+            {$inc: {'likes': 1} }
+        )
+        likes = post.get('likes');
+        console.log("likes", likes)
+    } catch (err) {
+        return res.status(400).json({
+            message: err.message
+        })
+    }
     try {
         await post.save()
     } catch (err) {
         res.status(400).json({
             message: err.message
         })
     }
+    res.status(201).json({
+        message: "incremented post like",
+        post
+    })
 }
 
 exports.getPostLikes = async function (req, res, next) {
-    const { authorEmail, post_id } = req.body 
-    const posts = await postModel.find({
-        authorEmail: authorEmail,
-        _id: post_id,
-    })
+    const { post_id } = req.body
+    // console.log(post_id);
+    try {
+        let post = await postModel.findById(post_id);
+        likes = post.get('likes');
+        console.log('likes', likes)
+    } catch (err) {
+        return res.status(400).json({
+            message: err.message
+        })
+    }
+
     res.status(200).json({
-        message: "Posts successfully updated.",
-        posts
+        message: "Get post likes",
+        likes
     })
 }
+exports.addPostComment = async function (req, res) {
+    const { authorEmail, post_id, comment } = req.body 
+    try{
+        let post = await postModel.findById(post_id)
+        await post.get('comments').push({
+            authorEmail: authorEmail,
+            body: comment,
+            date: new Date(),
+        })
+        await post.save()
+    } catch (err) {
+        return res.status(400).json({
+            message: err.message
+        })
+    }
 
-exports.getPostComments = async function (req, res, next) {
-    res.status(503).json({
-        message: "Not implemented yet",
+    res.status(201).json({
+        message: "Add Comments",
     })
 }
 
-exports.addPostComment = async function (req, res, next) {
-    res.status(503).json({
-        message: "Not implemented yet",
+exports.getPostComments = async function (req, res, next) {
+    const {post_id} = req.body
+    try {
+        let post = await postModel.findById(post_id);
+        comments = post.get('comments');
+        console.log('comments', comments)
+    } catch (err) {
+        return res.status(400).json({
+            message: err.message
+        })
+    }
+
+    res.status(200).json({
+        message: "Get post comments",
+        comments
     })
 }
 
 
-exports.savePost = async function (req, res, next) {
+exports.savePost = async function (req, res) {
     // add postid to saved posts field for student + club
-    res.status(503).json({
-        message: "Not implemented yet",
+    const {email, post_id} = req.body
+
+    try {
+        let user = await studentModel.findOne({ email })
+        user.savedPosts.push(post_id)
+        await user.save()
```

Comment:
Could you extend the savePost and getSavedPosts functionality to clubs too, by accepting a string in the query that lets you know if the request is being made by a student or club, and accordingly performing this exact same operation on clubModel if it's a club, and studentModel if it's a student?

Reply:
Also, don't forget to update the README!